core/src/broadcast_stage.rs

```diff
@@ -342,33 +343,33 @@ impl BroadcastStage {
         retransmit_slots_receiver: &RetransmitSlotsReceiver,
         socket_sender: &Sender<(Arc<Vec<Shred>>, Option<BroadcastShredBatchInfo>)>,
     ) -> Result<()> {
-        let timer = Duration::from_millis(100);
+        let mut sel = Select::new();
+        sel.recv(retransmit_slots_receiver);
+        let _ = sel.ready_timeout(Duration::from_millis(100))?;
 
         // Check for a retransmit signal
-        let mut retransmit_slots = retransmit_slots_receiver.recv_timeout(timer)?;
-        while let Ok(new_retransmit_slots) = retransmit_slots_receiver.try_recv() {
-            retransmit_slots.extend(new_retransmit_slots);
-        }
-
-        for (_, bank) in retransmit_slots.iter() {
-            let slot = bank.slot();
+        while let Ok(new_retransmit_slot) = retransmit_slots_receiver.try_recv() {
```

Comment:
`Select::try_ready` might return with success spuriously.
I suggest do instead something like:
```rust
const RECV_TIMEOUT: Duration = Duration::from_millis(100);
let retransmit_slots =
    std::iter::once(retransmit_slots_receiver.recv_timeout(RECV_TIMEOUT)?)
        .chain(retransmit_slots_receiver.try_iter());

for new_retransmit_slot in retransmit_slots {
    ...
```

Reply:
heh, and this is why you read the docs. Thanks for the suggestion!